### PR TITLE
test: rebuild unit tests in db_test.go(issue#374, task 4)

### PR DESCRIPTION
**Pull request by bigboss2063**
This is a **WIP** pr.

I made some modifications to have each test case use its own database instance, and I encapsulated some database operations to avoid redundant code. Is this refactoring feasible?

### CODE REVIEW

**Comment by elliotchenzichang** [comment]
Great progress! Looking forward to the rest of the code changes.

**Comment by bigboss2063** [comment]
Okay, thanks.

**Comment by bigboss2063** [comment]
I removed some non-reusable codes and encapsulated the utility functions that can be used in other tests.

File: db_test.go [hunk 1]
```diff
@@ -34,6 +33,71 @@ var (
...
+
+func removeDir(dir string) {
+	if err := os.RemoveAll(dir); err != nil {
+		panic(err)
+	}
+}
+
+func runNutsDBTest(t *testing.T, opts *Options, test func(t *testing.T, db *DB)) {
+	if opts == nil {
+		opts = new(Options)
```
Comment (elliotchenzichang) [inline db_test.go:52]: Just confusing, Finally, the opts value is DefaultOptions, why new an Options Object here?

File: db_test.go [hunk 2]
```diff
@@ -34,6 +33,71 @@ var (
...
+		*opts = DefaultOptions
+	}
+	if opts.Dir == "" {
+		opts.Dir = "/tmp/nutsdb-test"
```
Comment (elliotchenzichang) [inline db_test.go:56]: Maybe making this field a const value will be better. It can be used by other test functions.

**Review by elliotchenzichang — COMMENTED** [review]
General LGTM! Just left some comments! Thanks!

File: db_test.go (hunk 2 above)
Comment (bigboss2063) [reply db_test.go:56]: Okay

File: db_test.go (hunk 1 above)
Comment (bigboss2063) [reply db_test.go:52]: Because DefaultOptions is not a pointer, so I new an Options Object, Maybe I should use `opts = &DefaultOptions` to replace it.

**Comment by elliotchenzichang** [comment]
Merged. Thanks for this great contribution.

**Pull request merged**